src/compiler/gcc.rs

```diff
@@ -331,6 +332,7 @@ pub fn compile<T>(creator: &T,
     cmd.arg("-c")
         .arg(&parsed_args.input)
         .arg("-o").arg(&out_file)
+        .args(&parsed_args.preprocessor_args)
         .args(&parsed_args.common_args)
```

Comment:
I wonder if it would make sense to just use the original arguments. IIRC that's what the python implementation was doing.

FWIW, I dug in the code history, and this problem has existed ever since the rust implementation has had the "retry on error building the preprocessed output" code path for clang (092417b4)